### DISCUSSION TITLE
DDEV on windows with Docker in WSL2 without docker desktop

### DISCUSSION THREAD

**Discussion by grifmo** [ddev/ddev#5744] [Q&A]
Hi, I wonder if anyone could shed some light onto an issue I have. 

I have installed docker and DDEV using the WSL2 install script at https://raw.githubusercontent.com/ddev/ddev/master/scripts/install_ddev_wsl2_docker_inside.ps1

I got some errors along the way, when the script was running some of the wsl -u root -e bash commands, but I could run those within the wsl ubuntu shell without a problem, so eventually all of the commands in the script were run.

However. My understanding is that I should be able to run ddev from within windows terminal without entering the Ubuntu shell. This refuses to work. I always get the message:

`level=fatal msg="\x1b[31mUnable to get Docker context: unable to run 'docker context inspect' - please make sure Docker client is in path and up-to-date: exec: \"docker\": executable file not found in %PATH%\x1b[0m"
`

I have created aliases in my powershell profile, so I can run a docker command from within the terminal and it is passed to wsl which works perfectly. The surprising thing is that when I run ddev from within the ubuntu shell, I get the same message:

```
The hostname jondevtest.ddev.site is not currently resolvable, trying to add it to the hosts file
Unable to run ddev.exe, please check it on Windows side; err=exit status 100; output=time="2024-01-25T21:54:05+01:00" level=warning msg="Could not get Docker client. Is Docker running? Error: invalid endpoint"

Failed to start jondevtest: ddev.exe is not installed on the Windows side, please install it with 'choco install -y ddev'. It is used to manage the Windows hosts file
```
ddev is definitively installed in windows, it's just it will not recognize the docker aliases I have, nor does it find the docker host in wsl. I was fairly sure that I would be able to do this without installing docker desktop. Am I wrong?

**Comment by rfay** [comment]
Your understanding is incorrect. You run ddev commands in the wsl distro, in Ubuntu.

**Reply by grifmo** [reply]
that would be fine except for the message I get when i run ddev in ubuntu:

```
The hostname jondevtest.ddev.site is not currently resolvable, trying to add it to the hosts file
Unable to run ddev.exe, please check it on Windows side; err=exit status 100; output=time="2024-01-25T21:54:05+01:00" level=warning msg="Could not get Docker client. Is Docker running? Error: invalid endpoint"

Failed to start jondevtest: ddev.exe is not installed on the Windows side, please install it with 'choco install -y ddev'. It is used to manage the Windows hosts file
```

**Reply by rfay** [reply]
Your problem is that DNS is not working properly. Most common reason is https://ddev.readthedocs.io/en/latest/users/usage/troubleshooting/#dns-rebinding-prohibited-mostly-on-fritzbox-routers

**Reply by rfay** [reply]
But you also didn't get ddev installed properly on the Windows side, which is only installed there for this class of problem. It can edit the hosts file.


I recommend running the install script again.
